lib/ClientException.php

```diff
@@ -0,0 +1,121 @@
+<?php
+
+/**
+ * HTTP Client Exception Handler
+ *
+ * @author    Nurlan Mukhanov <nurike@gmail.com>
+ * @copyright 2019 Nurlan Mukhanov
+ * @license   https://opensource.org/licenses/MIT The MIT License
+ * @link      http://packagist.org/packages/sendgrid/php-http-client
+ */
+
+namespace SendGrid;
+
+use Exception;
+
+class ClientException extends Exception
+{
+    /**
+     * @var ClientError[] $errors
+     */
+    protected $errors;
+
+    /**
+     * @var int $httpStatus HTTP status code
+     */
+    protected $httpStatus;
+
+    /**
+     * @var array $fullTrace Full Exception trace
+     */
+    protected $fullTrace;
+
+    /**
+     * @var array $shortTrace Place where send method was called
+     */
+    protected $shortTrace;
+
+    /**
+     * ClientException constructor.
+     *
+     * @param string $message
+     * @param int    $httpStatus
+     * @param int    $code
+     */
+    public function __construct($message = "unhandled body response", $httpStatus = 0, $code = 0)
+    {
+        $this->httpStatus = $httpStatus;
+
+        $errors = null;
+
+        // If message is set
+        if($message) {
```

Comment:
To check whether the `$message` variable is set, I think it should use `isset` function to do this.